repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java

```diff
@@ -121,7 +131,190 @@ public void retryTask(String taskGuid) throws AtlasBaseException {
         setEncodedProperty(atlasVertex, Constants.TASK_STATUS, AtlasTask.Status.PENDING);
         int attemptCount = atlasVertex.getProperty(Constants.TASK_ATTEMPT_COUNT, Integer.class);
         setEncodedProperty(atlasVertex, Constants.TASK_ATTEMPT_COUNT, attemptCount+1);
+    }
+
+    @Override
+    @GraphTransaction
+    public List<AtlasTask> createAtlasTasks(List<AtlasTask> tasks) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("createAtlasTasks");
+        List<String> supportedTypes = new ArrayList<>();
+        // Get all the supported task type dynamically
+        Reflections reflections = new Reflections("org.apache.atlas.repository.store.graph.v2.tasks", new SubTypesScanner(false));
+        Set<Class<? extends TaskFactory>> classes = reflections.getSubTypesOf(TaskFactory.class);
+        for (Class<?> clazz : classes) {
+            List<Class<?>> interfaces = Arrays.asList(clazz.getInterfaces());
+            if (interfaces.contains(TaskFactory.class)) {
+                try {
+                    Field field = clazz.getField("supportedTypes");
+                    List<String> supportedTaskTypes = (List<String>) field.get(null);
+                    supportedTypes.addAll(supportedTaskTypes);
```

Comment:
Why fetching from reflections? the scope is limited as there are only a couple of implementations, can we populate this directly using lists in both implementation?

Reply:
Any Downside of this?

Reply:
Bit of performance issue but it shouldn't affect much I think

Reply:
I suggest removing the reflection to prepare allow types list, I do not think we will use this in the near future, we can edit to add to the list when needed.
Also missing this in the future will not cause any adverse effect other than an error message

Reply:
Yep gotcha!